File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESurfaceVariantLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceVariantLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl5
            AppTheme.STANDARD -> R.color.cstl5
            AppTheme.BLUE -> R.color.cbl5
            AppTheme.GREEN -> R.color.cgl5
            AppTheme.RED -> R.color.crl5
            AppTheme.NOTHING -> R.color.cnol5
        }
    }
}

File: lib/layer2_core/light/e_surface_variant_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESurfaceVariantLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl5;
      case 'standard': return cstl5;
      case 'blue': return cbl5;
      case 'green': return cgl5;
      case 'red': return crl5;
      case 'nothing': return cnol5;
      default: return cstl5;
    }
  }
}
